File: 1_two_sum/1_two_sum.go
package main

func twoSum(nums []int, target int) []int {
	complements := make(map[int]int)

	for i := 0; i < len(nums); i++ {
		complement, ok := complements[nums[i]]
		if ok {
			return []int{complement, i}
		}

		complements[target-nums[i]] = i
	}

	return nil
}

File: 1_two_sum/1_two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        complements = {}
        
        for i in range(len(nums)):
            complement = complements.get(nums[i])

            if complement != None:
                return [complement, i]
            
            complements[target - nums[i]] = i